app/scripts/metamask-controller.js

```diff
@@ -174,6 +175,23 @@ export default class MetamaskController extends EventEmitter {
       initState: initState.MetaMetricsController,
     });
 
+    this.gasFeeController = new GasFeeController({
+      interval: 10000,
+      messenger: controllerMessenger,
```

Comment:
It's rather important to pass in a restricted messenger here, not the global one. See the `currencyRateMessenger` for an example of this. This would fail to compile if this were in TypeScript.

Reply:
Resolved :D